command/resource/list-grpc/list_test.go

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package list
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/mitchellh/cli"
+
+	"github.com/hashicorp/consul/agent"
+	"github.com/hashicorp/consul/sdk/freeport"
+	"github.com/hashicorp/consul/testrpc"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/hashicorp/consul/command/resource/apply-grpc"
+)
+
+func TestResourceListCommand(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	t.Parallel()
+	availablePort := freeport.GetOne(t)
+	a := agent.NewTestAgent(t, fmt.Sprintf("ports { grpc = %d }", availablePort))
+	testrpc.WaitForTestAgent(t, a.RPC, "dc1")
+	t.Cleanup(func() {
+		a.Shutdown()
+	})
+
+	applyCli := cli.NewMockUi()
+
+	applyCmd := apply.New(applyCli)
+	code := applyCmd.Run([]string{
+		"-f=../testdata/demo.hcl",
+		fmt.Sprintf("-grpc-addr=127.0.0.1:%d", availablePort),
+		"-token=root",
+	})
+	require.Equal(t, 0, code)
+	require.Empty(t, applyCli.ErrorWriter.String())
+	require.Contains(t, applyCli.OutputWriter.String(), "demo.v2.Artist 'korn' created.")
+
+	cases := []struct {
+		name      string
+		output    string
+		extraArgs []string
+	}{
+		{
+			name:   "sample output",
+			output: "\"name\": \"korn\"",
+			extraArgs: []string{
+				"demo.v2.Artist",
+				"-namespace=default",
+				"-peer=local",
+				"-partition=default",
+			},
+		},
+		{
+			name:   "sample output",
+			output: "\"name\": \"korn\"",
+			extraArgs: []string{
+				"demo.v2.Artist",
+				"-p=korn",
+				"-namespace=default",
+				"-peer=local",
+				"-partition=default",
+			},
+		},
```

Comment:
nit: why do these test cases have the same name?

Reply:
good catch, I intend to test the name prefix but forgot to rename the test name. just updated